frontend/src/scenes/dashboard/dashboardLogic.tsx

```diff
@@ -1015,35 +1032,29 @@ export const dashboardLogic = kea<dashboardLogicType>([
             let refreshesFinished = 0
             const totalResponseBytes = 0
 
-            // array of functions that reload each item
+            // array of functions that reload each insight
             const fetchItemFunctions = insightsToRefresh.map((insight) => async () => {
-                const queryId = `${dashboardQueryId}::${uuid()}`
+                // dashboard refresh or insight refresh will have been triggered first
+                // so we should have a query_id to poll for
+                const queryId = insight.query_status?.id
                 const queryStartTime = performance.now()
 
                 try {
                     breakpoint()
-                    if (insight.query_status) {
-                        await pollForResults(insight.query_status.id, false, methodOptions)
-                        const currentTeamId = values.currentTeamId
-                        // TODO: Check and remove - We get the insight again here to get everything in the right format (e.g. because of result vs results)
-                        const polledInsight = await getSingleInsight(
-                            currentTeamId,
-                            insight,
-                            dashboardId,
-                            queryId,
-                            'force_cache',
-                            methodOptions
-                        )
-                        dashboardsModel.actions.updateDashboardInsight(polledInsight)
+                    if (queryId) {
+                        const statusResponse = await pollForResults(queryId, false, methodOptions)
+                        // retain existing metadata but update result
+                        const updatedInsight = { ...insight, result: statusResponse.results?.results }
```

Comment:
This is alright for the temporary filter update... but what now about real updates (think async query is finished and new results come in - with updated timestamps in metadata)?

Reply:
Thanks for pointing out! I missed the attributes like `last_refresh` that get updated after the async query completes. This doesn't seem to affect most UI flows but might be needed on the insight page to show the refresh button.
 
In general, what I noticed is that even for async refresh, the metadata for the insight will already have been updated before calling `updateInsightResults` as `loadDashboard` sets the [updated dashboard](https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/dashboard/dashboardLogic.tsx#L246) and then we use the [updated metadata](https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/dashboard/dashboardLogic.tsx#L641) in this function

Will try to think of a way to prevent another load of the insight

Reply:
> the metadata for the insight will already have been updated before calling updateInsightResults as loadDashboard sets the [updated dashboard](https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/dashboard/dashboardLogic.tsx?rgh-link-date=2024-08-13T10%3A55%3A08Z#L246) and then we use the [updated metadata](https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/dashboard/dashboardLogic.tsx?rgh-link-date=2024-08-13T10%3A55%3A08Z#L641) in this function

But for insights that calculate and are polling for e.g. a minute, the metadata is only then updated when the result is finished calculating. So it cannot be already in the initial load dashboard response?

Reply:
Yes, you're right, that's what I had missed in my initial changes.
I've put the `force_cache` insight call back in for now.
Will address that in a different PR as this PR's grown much bigger now that I'm combining the filter preview with the 'edit mode' 😅